plugins/base/src/main/kotlin/renderers/gfm/GfmRenderer.kt

```diff
@@ -0,0 +1,185 @@
+        if(node.ordered) {
+            buildListItem(node.children, pageContext, "${node.start}.")
+        } else {
+            buildListItem(node.children, pageContext, "*")
+        }
+    }
+
+    override fun StringBuilder.buildNewLine() {
+        this.append("  \n")
+    }
+
+    fun StringBuilder.buildParagraph() {
+        this.append("\n\n")
+    }
+
+    override fun StringBuilder.buildResource(node: ContentEmbeddedResource, pageContext: ContentPage) {
+        this.append("Resource")
+    }
+
+    override fun StringBuilder.buildTable(node: ContentTable, pageContext: ContentPage) {
+
+        val size = node.children.firstOrNull()?.children?.size ?: 0
+
+        buildParagraph()
+
+        if(node.header.size > 0) {
+            node.header.forEach {
+                it.children.forEach {
+                    append("| ")
+                    it.build(this, pageContext)
+                }
+                append("|\n")
+            }
+        } else {
+            append("| ".repeat(size))
+            if(size > 0) append("|\n")
+        }
+
+        append("|---".repeat(size))
+        if(size > 0) append("|\n")
+
+
+        node.children.forEach {
+            it.children.forEach {
+                append("| ")
+                it.build(this,  pageContext)
+            }
+            append("|\n")
+        }
+
+        buildParagraph()
+    }
+
+    private fun String.escapeIllegalCharacters() = this
+        .replace("<", "\\<")
+        .replace(">", "\\>")
+
+    override fun StringBuilder.buildText(textNode: ContentText) {
+        val decorators = decorators(textNode.style)
+        this.append(decorators)
+        this.append(textNode.text.escapeIllegalCharacters())
+        this.append(decorators.reversed())
+    }
+
+    override fun StringBuilder.buildNavigation(page: PageNode) {
+        locationProvider.ancestors(page).asReversed().forEach { node ->
+            append("/")
+            if (node.isNavigable) buildLink(node, page)
+            else append(node.name)
+        }
+        buildParagraph()
+    }
+
+    override fun buildPage(page: ContentPage, content: (StringBuilder, ContentPage) -> Unit): String {
+        val builder = StringBuilder()
+        content(builder, page)
+        return builder.toString()
+    }
+
+    override fun buildError(node: ContentNode) {
+        println("Error")
+    }
+
+    private fun decorators(styles: Set<Style>): String {
+        val decorators = StringBuilder()
+        styles.forEach {
+            when(it) {
+                TextStyle.Bold          -> decorators.append("**")
+                TextStyle.Italic        -> decorators.append("*")
+                TextStyle.Strong        -> decorators.append("**")
+                TextStyle.Strikethrough -> decorators.append("~~")
+                else                    -> Unit
+            }
+        }
+        return decorators.toString()
+    }
```

Comment:
```suggestion
    private fun decorators(styles: Set<Style>) = StringBuilder().let { decorators ->
        styles.forEach {
            when (it) {
                TextStyle.Bold -> decorators.append("**")
                TextStyle.Italic -> decorators.append("*")
                TextStyle.Strong -> decorators.append("**")
                TextStyle.Strikethrough -> decorators.append("~~")
                else -> Unit
            }
        }
    }.toString()
```
Looks better IMHO, also please use the formatter

Reply:
It may look better but once again it's not equal, cause `let` returns kotlin.Unit ('`forEach {}`), and whole result is `"kotlin.Unit"`

Reply:
It may look better but once again it's not equal, cause `let` returns kotlin.Unit (`forEach {}`), and whole result is `"kotlin.Unit"`

Reply:
Of course it can be mended with returning at the end of lambda decorators

Reply:
It may look better but once again it's not equal, cause `let` returns lambda result, which is `kotlin.Unit` (`forEach {}`). It can of course be easily mended by changing scope function to apply/also.